### DISCUSSION TITLE
MathML example does not work

### DISCUSSION THREAD

**Discussion by Parlazo** [wxMaxima-developers/wxmaxima#1656] [Q&A]
If i try this example

```
load("alt-display.mac")$
set_alt_display(2,mathml_display)$

x/(x^2+y^2) = 1;
<math xmlns="http://www.w3.org/1998/Math/MathML"> <mi>mlabel</mi> 
<mfenced separators=""><msub><mi>%o</mi> <mn>5</mn></msub> 
<mo>,</mo><mfrac><mrow><mi>x</mi> </mrow> <mrow><msup><mrow>
<mi>y</mi> </mrow> <mn>2</mn> </msup> <mo>+</mo> <msup><mrow>
<mi>x</mi> </mrow> <mn>2</mn> </msup> </mrow></mfrac> <mo>=</mo> 
<mn>1</mn> </mfenced> </math>;
```
from here https://maxima.sourceforge.io/docs/manual/maxima_175.html

i get a popup window with a red warning symbol and an ok button that tells me some thing about malformed token on line xy

the output is some thing like 

```
 <math<xmlns="http://www.w3.org/1998/Math/MathML"> <mi><mlabel<<mi> 
 <mfenced<separators=""><msub><<mi><%o<<mi> <mn><19<<mn><<msub> 
 <mo><,<<mo><mfrac><<mrow><<mi><x<<mi> <<mrow> <mrow><<msup><<mrow><
 <mi><y<<mi> <<mrow> <mn><2<<mn> <<msup> <mo><+<<mo> <msup><<mrow><
 <mi><x<<mi> <<mrow> <mn><2<<mn> <<msup> <<mrow><<mfrac> <mo><=<<mo> 
 <mn><1<<mn> <<mfenced> <<math>
```

some how the `<` seem to be misplaced or doubled

how to fix this ... or is this a bug?

**Comment by daute** [comment]
It is indeed a bug in wxMaxima, thanks for the report.
I tried a `git bisect` session and found the following commit as the cause:

commit ffc6b093942ac37d338a912a8935086a0d7185ca

Date:   Tue Aug 4 09:11:51 2020 +0200
    A faster method to find XML tagsffc6b093942ac37d338a912a8935086a0d7185ca in maxima's data

Maybe @gunterkoenigsmann knows how to fix it or if we should revert it (reverting is maybe not easy for a commit nearly 2 years ago...)

**Comment by gunterkoenigsmann** [comment]
Debugged it a bit: Maxima by default uses ASCII Art for drawing 2D equations. One can tell it to display them in 1D - or tell the alt-display package to do some fancy stuff like outputting Maths in MathML - or in the XML dialect wxMaxima uses since it contains both all the information needed for displaying nice 2D maths and for converting back every displayed object into maxima code.

wxMaxima understands a subset of the MathML maxima can generate and display it as nice 2D maths. My plans are to make wxMaxima completely understand MathML so we no more need to define our own XML code for transferring the maths. Before that is done https://sourceforge.net/p/maxima/feature-requests/165/ needs to be resolved, though, as the MathML from maxima currently does include enough information to be able to display nice maths, but not enough information for being sure what maxima commands that maths corresponds to.

**Comment by gunterkoenigsmann** [comment]
Filed the bug behind this question as https://github.com/wxMaxima-developers/wxmaxima/issues/1660.
